msal/application.py

```diff
@@ -79,6 +79,37 @@ def extract_certs(public_cert_content):
     return [public_cert_content.strip()]
 
 
+def _merge_claims_and_capabilities(capabilities, claims):
+    if claims and capabilities:
+        client_capabilities_dict = {
+            "access_token": {
+                "xms_cc": {
+                    "values": capabilities
+                }
+            }
+        }
+        claims_dict = json.loads(claims)
+        for key in client_capabilities_dict:
+            claims_node = claims_dict.get(key, {})
+            if claims_node:
+                claims_node.update(client_capabilities_dict[key])
+            else:
+                claims_dict[key] = client_capabilities_dict[key]
+        return json.dumps(claims_dict)
+    elif claims:
+        return claims
+    elif capabilities:
+        client_capabilities_dict = {
+            "access_token": {
+                "xms_cc": {
+                    "values": capabilities
+                }
+            }
+        }
+        return json.dumps(client_capabilities_dict)
+    return None
```

Comment:
Utilize the full power of Python's dict to cut 30-ish lines into 6. :-)
```suggestion
def _merge_claims_and_capabilities(capabilities, claims):
    # Represent capabilities as {"access_token": {"xms_cc": {"values": capabilities}}}
    # and then merge/add it into incoming claims
    if capabilities:
        claims_dict = json.loads(claims) if claims else {}
        for key in ["access_token"]:  # We could add "id_token" if we'd decide to
            claims_dict.setdefault(key, {}).update(xms_cc={"values": capabilities})
        return json.dumps(claims_dict)
    return claims
```